innopoints/views/account.py

```diff
@@ -64,3 +92,71 @@ def change_balance(email):
             abort(400, {'message': 'Data integrity violated.'})
 
     return jsonify(balance=user.balance)
+
+
+@api.route('/account/timeline', defaults={'email': None})
+@api.route('/account/<email>/timeline')
+@login_required
+def get_timeline(email):
+    """Get the timeline of the account.
+    If the e-mail is not passed, return own timeline."""
+    if email is None:
+        user = current_user
+    else:
+        if not current_user.is_admin:
+            abort(401)
+        user = Account.query.get_or_404(email)
+
+    # pylint: disable=bad-continuation
+
+    applications = (
+        db.session
+            .query(Application.id.label('application_id'),
+                   Application.status.label('application_status'))
+            .add_column(Application.application_time.label('entry_time'))
+            .filter_by(applicant=user)
+            .join(Activity).add_columns(Activity.name.label('activity_name'),
+                                        Activity.id.label('activity_id'))
+            .join(Project).add_columns(Project.name.label('project_name'),
+                                       Project.id.label('project_id'))
+            .add_column((Application.actual_hours * Activity.reward_rate).label('reward'))
+    ).subquery()
+
+    purchases = (
+        db.session
+            .query(StockChange.id.label('stock_change_id'),
+                   StockChange.status.label('stock_change_status'),
+                   StockChange.time.label('entry_time'))
+            .filter_by(account=user)
+            .filter(StockChange.amount < 0)
+            .join(Variety).join(Product).add_columns(Product.id.label('product_id'),
+                                                     Product.name.label('product_name'),
+                                                     Product.type.label('product_type'))
+    ).subquery()
+
+    promotions = (
+        # pylint: disable=unsubscriptable-object
+        db.session
+            .query(Notification.payload['project']['id'].label('project_id'),
+                   Notification.payload['project']['name'].label('project_name'),
```

Comment:
The payload stores only ids IDs and not objects. It looks like:
```json
{
    "project_id": 5
}
```
It is the Marshmallow schema that actually loads the object.

Reply:
That would be true in the case of notifications, but here I thought that since I query the IDs anyway, might as well include the information right away since it is available in that same query.

Reply:
I mean that we do not store the objects inside the payload in the database, just the IDs.
The notification payload object stored in Postgres looks like `{ "project_id": 1 }` rather than `{ "project": { "id": 1 } }`